source/type/data.cpp

```diff
@@ -101,26 +106,65 @@ bool Data::load(const std::string& filename,
         LOG4CPLUS_INFO(logger, boost::format("stopTimes : %d nb foot path : %d Nombre de stop points : %d")
                        % pt_data->nb_stop_times()
                        % pt_data->stop_point_connections.size()
-                       % pt_data->stop_points.size()
-            );
-        if (chaos_database) {
-            fill_disruption_from_database(*chaos_database, *pt_data, *meta, contributors);
-        }
-        build_raptor(raptor_cache_size);
+                       % pt_data->stop_points.size());
     } catch(const wrong_version& ex) {
         LOG4CPLUS_ERROR(logger, "Cannot load data: " << ex.what());
         last_load = false;
+        return last_load;
     } catch(const std::exception& ex) {
         LOG4CPLUS_ERROR(logger, "Data loading failed: " << ex.what());
         last_load = false;
+        return last_load;
     } catch(...) {
         LOG4CPLUS_ERROR(logger, "Data loading failed");
         last_load = false;
+        return last_load;
 	}
+
+    // Load Disruption (optionnal)
+    if ((chaos_database) && (disruptions_corruption_detected == false)){
+        try {
+            fill_disruption_from_database(*chaos_database, *pt_data, *meta, contributors);
+        } catch(const std::exception& ex) {
+            LOG4CPLUS_ERROR(logger, "Data disruptions loading failed: " << ex.what());
+            disruption_error = true;
+            if (std::string(ex.what()) != "Unable to connect to chaos database"){
+                LOG4CPLUS_INFO(logger, "Reload Data without disruptions");
+                disruptions_corruption_detected = true;
+                last_load = false;
+                return last_load;
+            }
+        } catch(...) {
+            disruption_error = true;
+            LOG4CPLUS_WARN(logger, "Data disruptions loading failed");
+            last_load = false;
+            return last_load;
+        }
+    }
+
+    // Build Raptor Data
+    try {
+        build_raptor(raptor_cache_size);
+    } catch(const std::exception& ex) {
+        LOG4CPLUS_ERROR(logger, "Build data Raptor failed: " << ex.what());
+        last_load = false;
+    } catch(...) {
+        LOG4CPLUS_ERROR(logger, "Build data Raptor failed");
+        last_load = false;
+    }
+
     loading = false;
     return this->last_load;
 }
 
+bool Data::load_without_disruptions(const std::string& filename,
+                                        const std::vector<std::string>& contributors,
+                                        const size_t raptor_cache_size) {
+   disruptions_corruption_detected = true;
```

Comment:
Why not pass it as a parameter and remove the attribute?
Not sure you need it in data_manager (but maybe), and it kinda feels like using the same thing for 2 correlated but different purposes (the code is more complicated to understand):
* one is whether you have to load data -> looks more like a param
* the other is "did it work well?" -> looks more like a return value (or a status attribute like now).